docs/decisions/0003-learning-path-enrollment-api.rst

```diff
@@ -0,0 +1,298 @@
+#################################
+0003 Learning Path Enrollment API
+#################################
+
+******
+Status
+******
+
+**Draft**
+
+.. Standard statuses
+    - **Draft** if the decision is newly proposed and in active discussion
+    - **Provisional** if the decision is still preliminary and in experimental phase
+    - **Accepted** *(date)* once it is agreed upon
+    - **Superseded** *(date)* with a reference to its replacement if a later ADR changes or reverses the decision
+
+    If an ADR has Draft status and the PR is under review, you can either use the intended final status (e.g. Provisional, Accepted, etc.), or you can clarify both the current and intended status using something like the following: "Draft (=> Provisional)". Either of these options is especially useful if the merged status is not intended to be Accepted.
+
+*******
+Context
+*******
+
+Learner enrollments in Learning Paths are captured in the LearningPathEnrollment
+model. In this ADR, it is enhanced with necessary fields and API to perform
+enrollments and store relevant data.
+
+********
+Decision
+********
+
+1. LearningPathEnrollment Model
+===============================
+
+Introduce new fields to the ``LearningPathEnrollment`` model:
+
+#. ``is_active`` as a ``BooleanField`` that will indicate the enrolled/unenrolled
+   state of the learner.
+#. ``history`` as a `django-simple-history`_ ``HistoricalRecords`` field to
+   preserve the record of changes made on an enrollment.
+#. ``enrolled_at`` as a ``DateTimeField`` to record the timestamp of the
+   enrollment. This will be explicitly set/reset when a user is enrolled in the
+   Learning Path. It will hold the latest valid enrollment timestamp in case of
+   un-enrollment and re-enrollment.
+
+.. _django-simple-history: https://django-simple-history.readthedocs.io/en/latest/quick_start.html
+
+2. ManualEnrollmentAudit Model
+==============================
+
+Create a new Django model called ``ManualEnrollmentAudit`` to store metadata
+about the Learning Path enrollments created manually.
+
+**Reference:** LMS student course enrollment's `ManualEnrollmentAudit`_
+
+Model attributes
+
+* ``enrollment`` - ``ForeignKey`` field referencing ``LearningPathEnrollment``
+* ``enrolled_by`` - ``ForeignKey`` field referencing the ``User`` performing the
+  enrollment.
+* ``role`` - ``CharField`` to store the role of the user creating the enrollment.
+* ``enrolled_email`` - ``CharField`` to store the email of the learner.
+* ``state_transition`` - ``CharField`` to store the enrollment state transition.
+  This field will use the existing `TRANSITION_STATES`_ from the LMS.
```

Comment:
A small warning: if we don't use `edx-platform` dependencies in the models yet, let's leave it this way. Depending on the whole `edx-platform` for migrations can make testing and CI pipelines more complex.

Reply:
That's a good point. I wanted to make sure the transition states recorded here are consistent with the existing values. But it does come with the cost of increasing the complexity. We can revisit during the implementation and maybe copy the same values. We are re-implementing a lot of the course related models. So, it's probably fine to just copy over the values.